kube-runtime/src/utils/watch_ext.rs

```diff
@@ -0,0 +1,42 @@
+use crate::{
+    utils::{event_flatten::EventFlatten, stream_backoff::StreamBackoff},
+    watcher,
+};
+use backoff::backoff::Backoff;
+
+use futures::{Stream, TryStream};
+
+/// Extension trait for streams returned by [`watcher`] or [`reflector`]
+pub trait WatchStreamExt: Stream {
+    /// Apply a [`Backoff`] policy to a [`Stream`] using [`StreamBackoff`]
+    fn backoff<B>(self, b: B) -> StreamBackoff<Self, B>
+    where
+        B: Backoff,
+        Self: TryStream + Sized,
+    {
+        StreamBackoff::new(self, b)
+    }
+
+    /// Flatten a [`watcher`] stream into a stream of applied objects
+    ///
+    /// All Added/Modified events are passed through, and critical errors bubble up.
+    /// This is functionally equivalent to calling [`try_flatten_applied`] on a [`watcher`].
+    fn watch_applies<K>(self) -> EventFlatten<Self, K>
```

Comment:
Maybe it's just me, but `WatchStreamExt::watch_applies` and `WatchStreamExt::watch_touches` sounds kind of awkward.

`applied_objects` and `touched_objects`?

```rust
watcher(api, lp).applied_objects()
watcher(api, lp).touched_objects()

reflector(store, watcher(api, lp)).applied_objects()
trigger_with(watcher(api, lp).touched_objects(), |obj| {})
```

Or even just `applied` and `touched`?

Reply:
yeah, I personally like those better myself, even the short variants makes sense to me.
if not the short ones, then at the very least the `_objects` suffix is better than a watch prefix - which is almost implied by the trait.

Reply:
Raised a PR in #906 for this.